File: JAVA/PokerWorld/src/bluffinmuffin/poker/game/dealer/AbstractDealer.java
package bluffinmuffin.poker.game.dealer;

import bluffinmuffin.game.Card;
import bluffinmuffin.game.CardSet;
import bluffinmuffin.poker.game.PlayerInfo;

public abstract class AbstractDealer
{
    protected CardSet m_deck;
    
    public AbstractDealer()
    {
        freshDeck();
    }
    
    public abstract CardSet dealHoles(PlayerInfo p);
    
    public abstract CardSet dealFlop();
    
    public abstract Card dealTurn();
    
    public abstract Card dealRiver();
    
    public abstract void freshDeck();
}

File: C#/Com.Ericmas001.Game.Poker.DataTypes/AbstractDealer.cs
using System;
using System.Collections.Generic;
using System.Text;
using Com.Ericmas001.Games;

namespace Com.Ericmas001.Game.Poker.DataTypes
{
    public abstract class AbstractDealer
    {
        protected Stack<GameCard> m_Deck;
       
        public AbstractDealer()
        {
            FreshDeck();
        }

        public abstract GameCard[] DealHoles();
        public abstract GameCard[] DealFlop();
        public abstract GameCard DealTurn();
        public abstract GameCard DealRiver();

        public abstract void FreshDeck();
    }
}
